pkgs/test/stdenv/default.nix

```diff
@@ -8,13 +12,15 @@
 }:
 
 let
+  # `stdenv` won't have a `__bootPackages` if it's `setupScript` has been overriden
+  # `stdenv.override { setupScript = ./pkgs/stdenv/generic/setup.sh; }`
+  stdenvHasBootPackages = stdenv ? __bootPackages;
   # early enough not to rebuild gcc but late enough to have patchelf
-  earlyPkgs = stdenv.__bootPackages.stdenv.__bootPackages;
-  earlierPkgs = stdenv.__bootPackages.stdenv.__bootPackages.stdenv.__bootPackages.stdenv.__bootPackages.stdenv.__bootPackages;
+  earlyPkgs = if stdenvHasBootPackages then stdenv.__bootPackages.stdenv.__bootPackages else pkgs;
+  earlierPkgs = if stdenvHasBootPackages then stdenv.__bootPackages.stdenv.__bootPackages.stdenv.__bootPackages.stdenv.__bootPackages.stdenv.__bootPackages else pkgs;
   # use a early stdenv so when hacking on stdenv this test can be run quickly
-  bootStdenv = stdenv.__bootPackages.stdenv.__bootPackages.stdenv.__bootPackages.stdenv.__bootPackages.stdenv;
-  pkgsStructured = import pkgs.path { config = { structuredAttrsByDefault = true; }; inherit (stdenv.hostPlatform) system; };
-  bootStdenvStructuredAttrsByDefault = pkgsStructured.stdenv.__bootPackages.stdenv.__bootPackages.stdenv.__bootPackages.stdenv.__bootPackages.stdenv;
+  bootStdenv = if stdenvHasBootPackages then stdenv.__bootPackages.stdenv.__bootPackages.stdenv.__bootPackages.stdenv.__bootPackages.stdenv else stdenv;
```

Comment:
Seaprate pr should make it so we use `longestValidPathPrefix` and `attrByPath` here to support darwin